src/gc/impl/conservative/gc.d

```diff
@@ -2414,12 +2457,39 @@ struct Gcx
                 start = stop;
             }
 
-            markAll(nostack);
+            if (block)
+            {
+                markAll(nostack);
+            }
+            else
+            {
+                // fork now and sweep later
+                fflush(null); // avoid duplicated FILE* output
+                auto pid = fork();
+                assert(pid != -1); // TODO handle case
+                switch (pid)
+                {
+                    case -1: // fork() failed, retry without forking
+                        markProcPid = 0;
+                        return fullcollect(nostack, true);
+                    case 0: // child process
+                            markAll(nostack);
+                            _Exit(0);
+                            break; // bogus
+                    default: // the parent
```

Comment:
The `block` variable is not properly handled here. Is this intentional? See https://github.com/sociomantic-tsunami/tangort/blob/v1.x.x/src/gc/cdgc/gc.d#L952-L970

Reply:
That block of code is not entered if the block variable is set to true (line 2461) and the mark is done in the current process.
Anyway, I updated the code to take into consideration a shouldFork boolean